src/rfkill/rfkill.c

```diff
@@ -335,9 +335,9 @@ static int run(int argc, char *argv[]) {
                         break;
                 }
 
-                if (l < RFKILL_EVENT_SIZE_V1)
-                        return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read of struct rfkill_event: (%zd < %d)",
-                                               l, RFKILL_EVENT_SIZE_V1);
+                if ((size_t)l < RFKILL_EVENT_SIZE_V1) /* l cannot be < 0 here. Cast to fix -Werror=sign-compare */
+                        return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read of struct rfkill_event: (%zd < %lu)",
+                                               l, (long unsigned int)RFKILL_EVENT_SIZE_V1); /* Casting necessary to make different archs happy */
```

Comment:
`long unsigned int` -> `long unsigned` (or `unsigned long`?)

Also, we usually add a white space between cast and variable or value: `(long unsigned) RFKILL_EVENT_SIZE_V1`.

Reply:
fixed both